src/autopas/containers/verletListsCellBased/verletLists/traversals/VLListIterationTraversal.h

```diff
@@ -47,10 +47,19 @@ class VLListIterationTraversal : public TraversalInterface, public VLTraversalIn
   void initTraversal() override {
     auto &cells = *(this->_cells);
     if (dataLayout == DataLayoutOption::soa) {
-      size_t offset = 0;
-      for (auto &cell : cells) {
-        _functor->SoALoader(cell, _soa, offset);
-        offset += cell.size();
+      // First resize the SoA to the required number of elements to store. This avoids resizing successively the SoA in
+      // SoALoader.
+      size_t numParticles = 0;
+      for (size_t i = 0; i < cells.size(); ++i) {
+        numParticles += cells[i].size();
+      }
+      _soa.resizeArrays(numParticles);
+
+      AUTOPAS_OPENMP(parallel for)
+      for (size_t i = 0; i < cells.size(); ++i) {
+        const auto offset = std::accumulate(cells.begin(), cells.begin() + i, 0,
+                                            [](const auto &acc, const auto &cell) { return acc + cell.size(); });
+        _functor->SoALoader(cells[i], _soa, offset, /*skipSoAResize*/ true);
```

Comment:
Is the `std::accumulate` in the loop really the most efficient way? This is basically in O(N)^2. Why not do a `std::inclusive_scan` instead of the first loop?
If this is really the best then please add a comment in the code because otherwise this question will come up again and again :D
(same for all other files with the similar pattern)

Reply:
Thanks for the tip. I didn't know `std::inclusive_scan` yet. Since we no longer do the summation of the cell sizes in parallel, I was able to combine the numParticles and offset calculation again with `std::inclusive_scan`.
This brings a further little improvement in performance.
I ran the scenario from above again on my Xeon E5-2650 with 8 threads:

Average times of `initTraversal` over all 20 iterations

|                 | avg time initTraversal (ns) |
|-----------------|-----------------------------|
| SoA loading for-loop ([454d438](https://github.com/AutoPas/AutoPas/pull/798/commits/454d4385d8cade46d7a1a712fe41c83cb622e74d)) | 111737193                   |
| SoA loading inclusive_scan ([6f21f4f](https://github.com/AutoPas/AutoPas/pull/798/commits/6f21f4f91505afb3638f8463f57e78a6f4dde4fb)) | 85892502                   |

![inclusive_scan](https://github.com/AutoPas/AutoPas/assets/121112647/7571db6e-a2f5-4ff1-9547-82a3d112dbed)